lib/src/main/kotlin/com/swmansion/starknet/data/types/Responses.kt

```diff
@@ -160,20 +192,33 @@ data class SyncingResponse(
     override val highestBlockNumber: Int,
 ) : Syncing()
 
+sealed interface GetBlockWithResponse {
+    val timestamp: Int
+    val sequencerAddress: Felt
+    val parentHash: Felt
+    val l1GasPrice: ResourcePrice
+    val l1DataGasPrice: ResourcePrice
+    val l1DataAvailabilityMode: L1DAMode
+    val starknetVersion: String
+}
+
+sealed interface ProcessedBlock : GetBlockWithResponse {
```

Comment:
nit: let's add a comment that this is in fact a `BlockHeader`, or change the name altogether. 
The same goes for PendingBlock

Reply:
Makes sense, do you think it should be docstring or a comment for internal use? 

Just a little bit unsure about `BlockHeader` name, seems like this is valid way to define it in the spec, but is a bit less appropriate for an interface name, as we are not dealing with responses per se (where having "header" makes sense), but with objects that are converted from those responses

Reply:
This is an interface, and it's not unreasonable to expect at least some users to have a look at the spec first, so perhaps this should be in the docstring?

Reply:
Yeah, I agree, but the problem is that `ProcessedBlock` does not fully correspond to `BLOCK_HEADER`, as there's a `status` property that is not in the header, but is always included alongside it. 

Ofc I can mention this in the dosctring. Just not sure dosctring that contains

```
Corresponds to `BLOCK_HEADER` with additional `status` property"
```
is really helpful 😅

Reply:
wait, so the nodes return some additional field that is not in spec? 
Just something like "The class reflects a BLOCK_HEADER object in starknet-spec, with an additional status field as returned by some of the nodes" ? Although just having an information that this is indeed a BLOCK_HEADER object in spec shoudl be sufficient enough imo

Reply:
> wait, so the nodes return some additional field that is not in spec?

No, it is in the spec, just that it's returned not as a part of `BLOCK_HEADER`, but alongside it, i.e.
```
"allOf": [
    {
        "title": "Block status",
        "type": "object",
        "properties": {
            "status": {
                "title": "Status",
                "$ref": "#/components/schemas/BLOCK_STATUS"
            }
        },
        "required": [
            "status"
        ]
    },
    {
        "title": "Block header",
        "$ref": "#/components/schemas/BLOCK_HEADER"
    },
    {
        "title": "Block body with transaction hashes",
        "$ref": "#/components/schemas/BLOCK_BODY_WITH_TX_HASHES"
    }
]
```

> Although just having an information that this is indeed a BLOCK_HEADER object in spec should be sufficient enough imo

I suppose we can avoid minor inconsistency here

Reply:
wait, so this is not a BLOCK_HEADER, but BLOCK_WITH_TX_HASHES right?

Reply:
If you're referring to the interface in the sdk, you could say it's `BLOCK_WITH_TX_HASHES` without `BLOCK_BODY_WITH_TX_HASHES` or `BLOCK_HEADER` + `BLOCK_STATUS`

Reply:
ok, so what is the final plan?

Reply:
I went ahead and added reference to `BLOCK_HEADER` to the docstring. I don't think an extra status property does any harm. I'm not sure we necessary have to document it, as it seems to me it might only add more confusion.